### ISSUE TITLE
[Mailer] SES API Transport authentication not working on ap-southeast-2

### ISSUE THREAD

**Issue by badams** [symfony/symfony#35468]
**Symfony version(s) affected**: 4.4/5.0

**Description**  
When using the SesApiTransport on the ap-southeast-2 region and valid credentials requests will fail with the following error
```
Unable to send an email: The security token included in the request is invalid (code InvalidClientTokenId).
```

**How to reproduce**  
Configure the Mailer with AWS API transport with region (ses+api://AWS_ACCESS_KEY_ID: AWS_SECRET_ACCESS_KEY@default?region=ap-southeast-2) and attempt to use the transport to send mail.

**Possible Solution**  
My investigation seems to indicate that the issue is caused by the way the authorization signature is generated https://github.com/symfony/amazon-mailer/blob/master/Transport/SesApiTransport.php#L85
as the result from the official sdk is quite different.

**Additional context**  
As a work around I've developed a custom transport that relies on the aws-php-sdk package for, unfortunately the base Transport class does not allow the addition of custom Transports so I can not use `Transport::fromDsn(`ses+sdk:...`)` to instantiate my transport

**Comment by OskarStark** [comment]
If your expectation is correct, it wouldn’t work with any region, not just the one you mentioned. Is that true?

**Comment by badams** [comment]
Agreed @OskarStark, unless I'm doing something terribly wrong. I've not tried with any other region as it takes time to verify a domain, and setup IAM roles etc, though I might if it helps..

For now though I've put together this minimal example reproducing the issue:
https://github.com/badams/aws-ses-test

Its also worth noting that the same credentials I used to test the above script work fine using SES via the SDK and this transport I've developed to get around this issue:
https://github.com/badams/symfony-mailer-amazon-sdk

**Comment by OskarStark** [comment]
Thank you for the detailed feedback, the reproducer and the code of your custom transport 🙏

**Comment by hyde1** [comment]
I don't know if the issue is related but on my side the signature failed if there is one of the following characters in the AWS_SECRET_KEY: `+` or `/`

**Comment by nielsmouthaan** [comment]
> I don't know if the issue is related but on my side the signature failed if there is one of the following characters in the AWS_SECRET_KEY: `+` or `/`.
> The only solution I found is to generate new credentials until there is no such character in the secret.

I can confirm this as I had the same issue...

**Comment by badams** [comment]
> I don't know if the issue is related but on my side the signature failed if there is one of the following characters in the AWS_SECRET_KEY: `+` or `/`.
> The only solution I found is to generate new credentials until there is no such character in the secret.

Yea I was aware of that issue, #32021 but it seemed unrelated from my testing - this was related to the signature generation changing in newer versions of the AWS API

**Comment by PWalkow** [comment]
Will this fix that exist in 5.1 will be merged back to 4.4?

**Comment by quiquegarcia** [comment]
Have you tried to use ```ses+smtp``` protocol and generate SMTP credentials from SES Home -> SMTP Settings -> Generate My SMTP Credentials?
